docs/topics/media-pipeline.rst

```diff
@@ -392,6 +392,32 @@ See here the methods that you can override in your custom Files Pipeline:
 
 .. class:: FilesPipeline
 
+   .. method:: file_path(request, response, info)
+
+      This method is called once per downloaded item. It returns the
+      download path of the file originating from the specified
+      :class:`response <scrapy.http.Response>`.
+
+      In addition to ``response``, this method receives the original
+      :class:`request <scrapy.Request>` and
+      :class:`info <scrapy.pipelines.media.MediaPipeline.SpiderInfo>`.
+
+      You can override this method to customize the download path of each file.
+
+      For example, if file URLs end like regular paths (e.g.
+      ``https://example.com/a/b/c/foo.png``), you can use the following
+      approach to download all files into the ``files`` folder with their
+      original filenames (e.g. ``files/foo.png``)::
+
+        import os
+        from urllib.parse import urlparse
+
+         def file_path(self, request, response, info):
+             return 'files/' + os.path.basename(urlparse(request.url).path)
```

Comment:
to make it a bit more clear, could you please add some boilerplate, e.g. 

```py
# ...
class MyFilesPipeline(FilesPipeline):
    # ...
    def file_path(self, request, response, info):
```

Reply:
I’ve gone with the whole implementation, as it only required a few more lines.